src/zope/testrunner/interfaces.py

```diff
@@ -100,3 +100,60 @@ class ITestRunner(zope.interface.Interface):
 
     options = zope.interface.Attribute(
       "Provides access to configuration options.")
+
+
+class IMinimalTestLayer(zope.interface.Interface):
+    """A test layer.
+
+    This is the bare minimum that a ``layer`` attribute on a test suite
+    should provide.
+    """
+
+    __bases__ = zope.interface.Attribute(
+        "A tuple of base layers.")
+
+    __name__ = zope.interface.Attribute(
+        "Name of the layer")
+
+    __module__ = zope.interface.Attribute(
+        "Dotted name of the module that defines this layer")
+
+    def __hash__():
+        """A test layer must be hashable"""
```

Comment:
I wasn't sure if I should include \__eq__ in the interface:

- On one hand, to be properly hashable an object has to have \__eq__ and \__hash__ and the two have to be consistent with each other.

- On the other hand Python object instances (including classes themselves) are identity-comparable and hashable by default, so the user doesn't have to do anything (unless they decide to override \__eq__).

So I chose a compromise and documented one of the two, thus combining the downsides.

I think you're right, a reminder about \__eq__ would be good.  (And also a comment that the default implementations provided by Python suffice.)